scipy/signal/_signaltools.py

```diff
@@ -1565,25 +1560,13 @@ def medfilt(volume, kernel_size=None):
                       'zero-padded.',
                       stacklevel=2)
 
-    domain = np.ones(kernel_size, dtype=volume.dtype)
-
-    numels = np.prod(kernel_size, axis=0)
-    order = numels // 2
-
     if volume.dtype in [np.bool_, np.complex64, np.complex128, np.clongdouble,
-                        np.float16]:
+                        np.float16, np.object_, 'float128']:
```

Comment:
Note that you want to use `longdouble` as the dtype name, because that's the canonical one (`float128` may not exist, and then the alias `float96` will exist)